File: frontend/src/lib/types/lead.ts
import type { ClientCreateResult } from "@/lib/types/client";

export type LeadStatus = "NEW" | "CONTACTED" | "CONVERTED" | "CLOSED";

export interface Lead {
  uuid: string;
  name: string;
  email: string;
  phone: string | null;
  company: string | null;
  message: string | null;
  source: string | null;
  status: LeadStatus;
  notes: string | null;
  createdAt: string;
}

export interface LeadConvertResult {
  lead: Lead;
  client: ClientCreateResult;
}

File: backend/src/main/java/com/nfcplatform/lead/entity/Lead.java
package com.nfcplatform.lead.entity;

import com.nfcplatform.common.entity.Auditable;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "leads")
public class Lead extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, updatable = false, length = 36)
    private String uuid = UUID.randomUUID().toString();

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String email;

    private String phone;

    private String company;

    @Column(length = 2000)
    private String message;

    private String source;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private LeadStatus status = LeadStatus.NEW;

    @Column(length = 2000)
    private String notes;
}
